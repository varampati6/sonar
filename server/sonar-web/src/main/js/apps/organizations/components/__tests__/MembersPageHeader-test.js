/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import React from 'react';
import { shallow } from 'enzyme';
import MembersPageHeader from '../MembersPageHeader';

it('should render the members page header', () => {
  const wrapper = shallow(<MembersPageHeader />);
  expect(wrapper).toMatchSnapshot();
  wrapper.setProps({ loading: true });
  expect(wrapper.find('.spinner')).toMatchSnapshot();
});

it('should render the members page header with the total', () => {
  const wrapper = shallow(<MembersPageHeader total="5" />);
  expect(wrapper).toMatchSnapshot();
});

it('should render its children', () => {
  const wrapper = shallow(
    <MembersPageHeader loading={true} total="5">
      <span>children test</span>
    </MembersPageHeader>
  );
  expect(wrapper).toMatchSnapshot();
});
