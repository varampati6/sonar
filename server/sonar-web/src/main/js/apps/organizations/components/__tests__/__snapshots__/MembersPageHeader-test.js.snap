// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render its children 1`] = `
<header
  className="page-header"
>
  <i
    className="spinner"
  />
  <span>
    children test
  </span>
</header>
`;

exports[`should render the members page header 1`] = `
<header
  className="page-header"
/>
`;

exports[`should render the members page header 2`] = `
<i
  className="spinner"
/>
`;

exports[`should render the members page header with the total 1`] = `
<header
  className="page-header"
/>
`;
